lib/internal/vm.js

```diff
@@ -24,12 +24,36 @@ const {
   ERR_INVALID_ARG_TYPE,
 } = require('internal/errors').codes;
 
+/**
+ * Checks if the given object is a context object.
+ * @param {object} object - The object to check.
+ * @returns {boolean} - Returns true if the object is a context object, else false.
+ */
 function isContext(object) {
   validateObject(object, 'object', kValidateObjectAllowArray);
 
   return _isContext(object);
 }
 
+/**
+ * Compiles a function from the given code string.
+ * @param {string} code - The code string to compile.
+ * @param {string[]} [params] - An optional array of parameter names for the compiled function.
+ * @param {object} [options] - An optional object containing compilation options.
+ * @param {string} [options.filename=''] - The filename to use for the compiled function.
+ * @param {number} [options.columnOffset=0] - The column offset to use for the compiled function.
+ * @param {number} [options.lineOffset=0] - The line offset to use for the compiled function.
+ * @param {Buffer} [options.cachedData=undefined] - The cached data to use for the compiled function.
+ * @param {boolean} [options.produceCachedData=false] - Whether to produce cached data for the compiled function.
+ * @param {ReturnType<import('vm').createContext} [options.parsingContext=undefined] - The parsing context to use for
+ * the compiled function.
+ * @param {object[]} [options.contextExtensions=[]] - An array of context extensions to use for the compiled function.
+ * @param {import('internal/modules/esm/utils').ImportModuleDynamicallyCallback} [options.importModuleDynamically] -
+ * A function to use for dynamically importing modules.
+ * @returns {object} An object containing the compiled function and any associated data.
+ * @throws {TypeError} If any of the arguments are of the wrong type.
+ * @throws {ERR_INVALID_ARG_TYPE} If the parsing context is not a valid context object.
+ */
 function internalCompileFunction(code, params, options) {
```

Comment:
FYI I am changing the signature to be explicit arguments validated by the public version in https://github.com/nodejs/node/pull/50137 because it's already weird/slow that the internal implementation still has to verify arguments and also the internal version should allow more customization to, er, internal options (i.e. host defined options).

Reply:
So do you not want to merge this PR in? Or add updated typings to that PR?